src/test/java/com/browserstack/test/suites/TestBase.java

```diff
@@ -0,0 +1,119 @@
+package com.browserstack.test.suites;
+
+import com.browserstack.app.utils.OsUtils;
+import com.browserstack.local.Local;
+import com.browserstack.test.utils.ScreenshotListener;
+import org.json.simple.JSONObject;
+import org.json.simple.parser.JSONParser;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.chrome.ChromeDriver;
+import org.openqa.selenium.remote.CapabilityType;
+import org.openqa.selenium.remote.DesiredCapabilities;
+import org.openqa.selenium.remote.RemoteWebDriver;
+import org.openqa.selenium.support.ui.WebDriverWait;
+import org.testng.annotations.*;
+
+import java.io.FileReader;
+import java.net.URL;
+import java.nio.file.Path;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.UUID;
+import java.util.concurrent.TimeUnit;
+
+@Listeners({ScreenshotListener.class})
+public class TestBase {
+    // ThreadLocal gives the ability to store data individually for the current thread
+    protected static ThreadLocal<WebDriver> driver = new ThreadLocal<>();
+    private Local local;
+    protected WebDriverWait wait;
+    protected String driverLocation = Path.of(System.getProperty("user.dir"), "/src/test/resources/drivers").toString();
+
+    public WebDriver getDriver() {
+        return driver.get();
+    }
+
+    public String getGeolocation() {
+        return "";
+    }
+
+    @BeforeMethod
+    @Parameters(value = {"environment", "test", "env_cap_id"})
+    public void setUp(@Optional("local") String environment, @Optional("single") String test, @Optional("0") int env_cap_id) throws Exception {
+        JSONParser parser = new JSONParser();
+        JSONObject config = (JSONObject) parser.parse(new FileReader("src/test/resources/conf/capabilities/test_caps.json"));
+        String url = (String) config.get("application_endpoint");
+        if (environment.equalsIgnoreCase("local")) {
+            if (OsUtils.isMac()) {
+                System.setProperty("webdriver.chrome.driver", Path.of(driverLocation, "/chromedriver").toString());
+            }
+            if (OsUtils.isWindows()) {
+                System.setProperty("webdriver.chrome.driver", Path.of(driverLocation, "/chromedriver.exe").toString());
+            }
+            if (OsUtils.isUnix()) {
+                System.setProperty("webdriver.chrome.driver", Path.of(driverLocation, "/chromedriver").toString());
```

Comment:
Chromedriver doesn't existing in the driver folder on-prem test failing due to missing driver